### PR TITLE
spec: improve Web::PushSubscription branch coverage

### PR DESCRIPTION
**Pull request by at9596**
## Summary

Improves test coverage for `Web::PushSubscription` by adding specs for previously unexercised branches and class-level behavior.

## Changes

* Added edge case coverage for:

  * missing alerts configuration
  * unknown notification types
  * missing policy configuration
  * unknown policy values
* Added coverage for `.unsubscribe_for`
* Updated predicate expectations to use idiomatic RSpec matcher syntax (`be_pushable`)
* Improved branch coverage without changing application behavior

## Result

Brings the model closer to full line and branch coverage while keeping specs behavior-focused and aligned with RSpec style conventions.

### CODE REVIEW

File: spec/models/web/push_subscription_spec.rb [hunk 1]
```diff
@@ -41,15 +41,17 @@
       let(:policy) { 'all' }
 
       it 'returns true' do
-        expect(subject.pushable?(notification)).to be true
+        # Predicate matcher syntax automatically delegates to `pushable?`
```
Comment (mjankowski) [inline spec/models/web/push_subscription_spec.rb]: I would drop all these comment lines. This should be obvious even if you are not familiar with this rspec feature.

File: spec/models/web/push_subscription_spec.rb [hunk 2]
```diff
@@ -82,19 +84,88 @@
...
+
+    let(:access_token) do
+      Fabricate(
+        :accessible_access_token,
+        application: application,
+        resource_owner_id: user.id
+      )
+    end
+
+    before do
```
Comment (mjankowski) [inline spec/models/web/push_subscription_spec.rb]: Maybe use a `let!` instead of this before just to create the record?
